orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/EvaluateConditionTask.java

```diff
@@ -0,0 +1,105 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.tasks;
+
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.conditions.ConditionConfigurationProperties;
+import com.netflix.spinnaker.orca.conditions.Condition;
+import com.netflix.spinnaker.orca.conditions.ConditionSupplier;
+import com.netflix.spinnaker.orca.pipeline.WaitForConditionStage.WaitForConditionContext;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.time.Clock;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.netflix.spinnaker.orca.pipeline.WaitForConditionStage.WaitForConditionContext.*;
+
+@Component
+public class EvaluateConditionTask implements RetryableTask {
+  private static Logger log = LoggerFactory.getLogger(EvaluateConditionTask.class);
+  private final ConditionConfigurationProperties conditionsConfigurationProperties;
+  private final Optional<List<ConditionSupplier>> suppliers;
+  private final Clock clock;
+
+  @Autowired
+  public EvaluateConditionTask(
+    ConditionConfigurationProperties conditionsConfigurationProperties,
+    Optional<List<ConditionSupplier>> suppliers,
+    Clock clock
+  ) {
+    this.conditionsConfigurationProperties = conditionsConfigurationProperties;
+    this.suppliers = suppliers;
+    this.clock = clock;
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.MINUTES.toMillis(conditionsConfigurationProperties.getBackoffWaitMins());
+  }
+
+  @Override
+  public long getTimeout() {
+    return Integer.MAX_VALUE;
+  }
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    if (!conditionsConfigurationProperties.isEnabled() || !suppliers.isPresent()) {
+      log.warn("Conditions not configured. Skipping wait.");
+      stage.getContext().put("status", Status.SKIPPED);
+      return new TaskResult(ExecutionStatus.SUCCEEDED, stage.getContext());
+    }
+
+    final WaitForConditionContext waitForConditionContext = stage.mapTo(WaitForConditionContext.class);
+    if (waitForConditionContext.getStatus() == Status.SKIPPED) {
```

Comment:
when does this `if` evaluate to `true`? Everywhere `status` is set to `Status.SKIPPED` the task completes with `SUCCEEDED` so this execute method shouldn't run again?
I mostly ask, because I am still trying to wrap my head around the execution flow in general. I feel like everytime I got it - I find something that i don't understand

Reply:
`WaitForConditionContext.status` here is used as a marker. `status=SKIPPED` (when the user decides to skip the wait) would short-circuit the wait. Think of it as when a user decides to skip the remainder of `waitTime` in a wait stage. Happy to go over the general flow of tasks (retryable or not), stages, offline if you'd like.

Reply:
got it, that makes sense! thanks for the explanation
I assume the UI has a list of stages that can be skipped so that will need to be updated?